android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -609,6 +606,181 @@ constructor(
     }
   }
 
+  /**
+   * Validates the given Questionnaire Response using the SDK [QuestionnaireResponseValidator].
+   *
+   * @param questionnaire Questionnaire to use in validation
+   * @param questionnaireResponse QuestionnaireResponse to validate
+   * @param context Context to use in validation
+   */
+  fun isQuestionnaireResponseValid(
+    questionnaire: Questionnaire,
+    questionnaireResponse: QuestionnaireResponse,
+    context: Context
+  ): Boolean {
+    return try {
+      QuestionnaireResponseValidator.checkQuestionnaireResponse(
+        questionnaire,
+        questionnaireResponse
+      )
+      QuestionnaireResponseValidator.validateQuestionnaireResponse(
+        questionnaire,
+        questionnaireResponse,
+        context
+      )
```

Comment:
why do we need to do this both without and with context?

Reply:
Oh, it's just a parameter requirement from the SDK.
`validateQuestionnaireResponse` uses context to get a string resource for the validation results.
`checkQuestionnaireResponse ` doesn't do that so it doesn't need context.